addons/sale/models/mail_thread.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+class MailThread(models.AbstractModel):
+    _inherit = 'mail.thread'
+
+    def _track_finalize(self):
+        # In the case sale orders are in draft mode do not log changes on the chatter.
+        # This is thought for sale orders while in form view or when in the catalog,
+        # in these cases only one sale order is selected.
+        if self._name == 'sale.order' and len(self) == 1 and self.state == 'draft':
```

Comment:
To be honest, I feel like this is a hack for no real added value. Tracking values is standard in Odoo and I think it is better to stick on simple and reliable behavior, instead of adding magic. But that being said it is your choice to do it or not as it is localized on SO model.

Reply:
@tde-banana-odoo It is desired for sale orders to not log changes on the chatter. From your help from discord this looks like the best way. If there is a more clean solution to it I will happily take it :)